Source/Session.swift

```diff
@@ -192,6 +194,22 @@ open class Session {
         session.invalidateAndCancel()
     }
 
+    // MARK: - Cancellation
+
+    /// Cancel all active `Request`s, optionally calling a completion handler when complete.
+    ///
+    /// - Note: This is an asynchronous operation and does not block the creation of future `Request`s. Cancelled
+    ///         `Request`s may not cancel immediately due internal work, and may not cancel at all if they are close to
+    ///         completion when cancelled.
+    ///
+    /// - Parameter completion: Closure to be called when all `Request`s have been cancelled.
+    public func cancelAllRequests(completion: (() -> Void)? = nil) {
+        rootQueue.async {
+            self.activeRequests.forEach { $0.cancel() }
+            completion?()
```

Comment:
Should we hop back off `rootQueue` here to avoid clients hanging it up? If yes, should we provide a `queue` parameter that defaults maybe to `.main`?

Reply:
I was trying to match `URLSession` behavior here, but this is a better idea. Added in https://github.com/Alamofire/Alamofire/pull/2890/commits/39aaa58af41d5d423c60d21613c64a1f5b75d0bd